library/src/main/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionedRecyclerViewAdapter.java

```diff
@@ -793,6 +793,122 @@ void callSuperNotifyItemMoved(int fromPosition, int toPosition) {
+        int footerPosition = getFooterPositionInAdapter(tag);
+
+        callSuperNotifyItemInserted(footerPosition);
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemInserted} with the position of the section's
+     * footer in the adapter. Useful to be called after changing the visibility of the section's
+     * footer to visible with {@link Section#setHasFooter}.
+     *
+     * @param section a visible section of this adapter
+     */
+    public void notifyFooterInsertedInSection(Section section) {
+        int footerPosition = getFooterPositionInAdapter(section);
+
+        callSuperNotifyItemInserted(footerPosition);
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemRangeInserted} with the position of the section
+     * in the adapter. Useful to be called after changing the visibility of the section to visible
+     * with {@link Section#setVisible}.
+     *
+     * @param tag unique identifier of the section
+     */
+    public void notifySectionChangedToVisible(String tag) {
+        Section section = getSection(tag);
+
+        notifySectionChangedToVisible(section);
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemRangeInserted} with the position of the section
+     * in the adapter. Useful to be called after changing the visibility of the section to invisible
+     * with {@link Section#setVisible}.
+     *
+     * @param tag unique identifier of the section
+     * @param previousSectionPosition previous section position
+     */
+    public void notifySectionChangedToInvisible(String tag, int previousSectionPosition) {
+        Section section = getValidSectionOrThrowException(tag);
+
+        notifySectionChangedToInvisible(section, previousSectionPosition);
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemRangeInserted} with the position of the section
+     * in the adapter. Useful to be called after changing the visibility of the section to visible
+     * with {@link Section#setVisible}.
+     *
+     * @param section a visible section of this adapter
+     */
+    public void notifySectionChangedToVisible(Section section) {
+        if (!section.isVisible()) {
+            throw new IllegalStateException("This section is not visible.");
+        }
+
+        int sectionPosition = getSectionPosition(section);
+
+        int sectionItemsTotal = section.getSectionItemsTotal();
+
+        callSuperNotifyItemRangeInserted(sectionPosition, sectionItemsTotal);
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemRangeInserted} with the position of the section
+     * in the adapter. Useful to be called after changing the visibility of the section to invisible
+     * with {@link Section#setVisible}.
+     *
+     * @param section an invisible section of this adapter
+     * @param previousSectionPosition previous section position
+     */
+    public void notifySectionChangedToInvisible(Section section, int previousSectionPosition) {
```

Comment:
and as result this one should be under the corresponded `notifySectionChangedToInvisible(tag)`